src/csharp/Grpc.HealthCheck/HealthGrpc.cs

```diff
@@ -38,11 +38,22 @@ public interface IHealthClient
     }
 
     // server-side interface
+    [System.Obsolete("Service implementations should inherit from the generated abstract base class instead.")]
     public interface IHealth
     {
       Task<global::Grpc.Health.V1.HealthCheckResponse> Check(global::Grpc.Health.V1.HealthCheckRequest request, ServerCallContext context);
     }
 
+    // server-side abstract class
+    public abstract class HealthBase
```

Comment:
I think it would be less surprising, basically. It's distinctly odd to see an abstract and an interface with the same members, but no type relationship.

Personally I think it would be cleaner just to kill the interface entirely and immediately, but okay, I'm content with this otherwise.